larray/inout/csv.py

```diff
@@ -70,58 +70,133 @@ def read_csv(filepath_or_buffer, nb_axes=None, index_col=None, sep=',', headerse
     >>> read_csv(fname)
-    nat\\sex  M  F
-         BE  0  1
-         FO  2  3
-
-    Sort columns
+    a  b\c    c0    c1    c2
+    1   b0   0.0   1.0   2.0
+    1   b1   3.0   4.0   5.0
+    2   b0   nan   nan   nan
+    2   b1   9.0  10.0  11.0
+    3   b0  12.0  13.0  14.0
+    3   b1   nan   nan   nan
+    >>> # using argument 'fill_value', you can choose which value to set in cells
+    >>> # associated with missing label .
+    >>> read_csv(fname, fill_value=0)
+    a  b\c  c0  c1  c2
+    1   b0   0   1   2
+    1   b1   3   4   5
+    2   b0   0   0   0
+    2   b1   9  10  11
+    3   b0  12  13  14
+    3   b1   0   0   0
 
-    >>> read_csv(fname, sort_columns=True)
-    nat\\sex  F  M
-         BE  1  0
-         FO  3  2
-
-    Read array saved in "narrow" format (wide=False)
+    Specify the number of axes of the output array (useful when the name of the last axis is implicit)
 
-    >>> a.to_csv(fname, wide=False)
+    >>> fname = os.path.join(EXAMPLE_FILES_DIR, 'test2d_classic.csv')
+    >>> # let's take a look inside the CSV file.
+    >>> # The name of the second axis is missing.
     >>> with open(fname) as f:
     ...     print(f.read().strip())
-    nat,sex,value
-    BE,M,0
-    BE,F,1
-    FO,M,2
-    FO,F,3
-    >>> read_csv(fname, wide=False)
-    nat\\sex  M  F
-         BE  0  1
-         FO  2  3
+    a,b0,b1,b2
+    a0,0,1,2
+    a1,3,4,5
+    a2,6,7,8
+    >>> # read the array stored in the CSV file as it
+    >>> read_csv(fname)
+    a\{1}  b0  b1  b2
+       a0   0   1   2
+       a1   3   4   5
+       a2   6   7   8
+    >>> # using argument 'nb_axes', you can force the number of axes of the output array
+    >>> read_csv(fname, nb_axes=2)
+    a\{1}  b0  b1  b2
+       a0   0   1   2
+       a1   3   4   5
+       a2   6   7   8
 
-    Specify the number of axes of the output array (useful when the name of the last axis is implicit)
+    Read array saved in "narrow" format (wide=False)
 
-    >>> a.to_csv(fname, dialect='classic')
+    >>> fname = os.path.join(EXAMPLE_FILES_DIR, 'test3d_narrow.csv')
+    >>> # let's take a look inside the CSV file.
+    >>> # Here, data are stored in a 'narrow' format.
     >>> with open(fname) as f:
     ...     print(f.read().strip())
-    nat,M,F
-    BE,0,1
-    FO,2,3
-    >>> read_csv(fname, nb_axes=2)
-    nat\\{1}  M  F
-         BE  0  1
-         FO  2  3
+    a,b,c,value
+    1,b0,c0,0
+    1,b0,c1,1
+    1,b0,c2,2
+    1,b1,c0,3
+    1,b1,c1,4
+    1,b1,c2,5
+    2,b0,c0,6
+    2,b0,c1,7
+    2,b0,c2,8
+    2,b1,c0,9
+    2,b1,c1,10
+    2,b1,c2,11
+    3,b0,c0,12
+    3,b0,c1,13
+    3,b0,c2,14
+    3,b1,c0,15
+    3,b1,c1,16
+    3,b1,c2,17
+    >>> # to read arrays stored in 'narrow' format, you must pass wide=False to read_csv
+    >>> read_csv(fname, wide=False)
+    a  b\c  c0  c1  c2
+    1   b0   0   1   2
+    1   b1   3   4   5
+    2   b0   6   7   8
+    2   b1   9  10  11
+    3   b0  12  13  14
+    3   b1  15  16  17
+
+    Sort rows and columns
```

Comment:
Since the argument is self-explanatory in this case, I don't think it's worth to add examples for it.